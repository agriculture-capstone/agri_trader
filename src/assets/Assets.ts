export const images = {
    back: require('../../src/Assets/back_white_24dp.png'),
    menu: require('../../src/Assets/menu_white_24dp.png'),
    settings: require('../../src/Assets/settings_white_24dp.png'),
    account: require('../../src/Assets/account_white_24dp.png'),
};

export default images;
