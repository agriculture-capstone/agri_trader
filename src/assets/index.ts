export const icons = {
  account: require('../../src/assets/account_white_24dp.png'),    
  back: require('../../src/assets/back_white_24dp.png'),
  menu: require('../../src/assets/menu_white_24dp.png'),
  settings: require('../../src/assets/settings_white_24dp.png'),
};
