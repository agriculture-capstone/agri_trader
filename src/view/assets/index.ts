export const images = {
  logo: require('./boresha_circular_logo.png'),
  truck_icon: require('./truck.png'),
  logo_dark: require('./boresha_circular_logo_dark.png'),
};
