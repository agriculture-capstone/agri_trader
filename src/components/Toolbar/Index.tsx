import {
  StyleSheet,
  ToolbarAndroid,
  ToolbarAndroidAction,
} from 'react-native';
import * as React from 'react';
import { icons } from '../../assets/';

/**
  * Types of permitted left toolbar buttons
  */
export enum LeftButtonTypes {
  menu = 'menu',
  back = 'back',
}
/**
  * action menu properties
  */
export interface RightButtonType {
  title: string;
  icon?: any;
  action: () => void;
}

interface ToolbarProps {
  title: string;
  leftButtonType: LeftButtonTypes;
  rightButtons: RightButtonType[];
}
/**
 * Class representing a Toolbar
 * @extends React.Component<ToolbarProps, {}>
 */
export default class Toolbar extends React.Component<ToolbarProps, {}> {
  constructor(props: ToolbarProps) {
    super(props);
    this.onActionSelected = this.onActionSelected.bind(this);
  }
  /**
   * When a button is selected by a user, find its position in the actions
   * array and call specified function
   */
  public onActionSelected(position: number) {
    this.props.rightButtons[position].action();
  }
  /**
   * Render method for Toolbar
   */
  public render() {
    const actionArray = ToolbarAndroidAction[]();
    this.props.rightButtons.map(rightButton => {
      const action: ToolbarAndroidAction = {
        title: RightButton.title,
        show: 'ifRoom',
      };
      if (RightButton.icon) {
        action.icon = RightButton.icon;
      }
      actionArray.push(action);
    })
  private return(<
    ToolbarAndroid title= {
      this.props.title
    }
    titleColor="white"
    style={
      styles.toolbar
    }
    navIcon={(this.props.leftButtonType == LeftButtonTypes.menu) ? icons.menu : icons.back
    }
    onIconClicked={(this.props.leftButtonType == LeftButtonTypes.menu) ? () => { } : () => { }
    }
    onActionSelected={
      this.onActionSelected
    }
    actions={
      actionArray
    }
  />
  );
}
}

const styles = StyleSheet.create({
  toolbar: {
    height: 56,
    backgroundColor: '#4883da',
  },
});
