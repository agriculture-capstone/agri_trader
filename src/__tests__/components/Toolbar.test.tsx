// Toolbar.test.tsx
import 'react-native';
import * as React from 'react';
import Toolbar, {
  LeftButtonTypes, RightButtonType,
} from '../../components/Toolbar';
import * as renderer from 'react-test-renderer';
import { icons } from '../../assets';

describe('<Toolbar/>', function () {
  const TITLE = 'Agritrader';
  const BUTTON_TEMPLATE = 'button_';
  const MULTI_BUTTON_COUNT = 2;

  describe('rendering left button and 0 right buttons', function () {
    it('creates valid snapshot', function () {
      const component = renderer.create(
        <Toolbar
          title={TITLE}
          rightButtons={addRightButtons(0)}
          leftButtonType={LeftButtonTypes.menu}
        />,
      );
      expect(component).toMatchSnapshot();
    });
  });
  describe('rendering 1 right button', function () {
    it('creates valid snapshot', function () {
      const component = renderer.create(
        <Toolbar
          title={TITLE}
          rightButtons={addRightButtons(1)}
          leftButtonType={LeftButtonTypes.back}
        />,
      );
      expect(component).toMatchSnapshot();
    });
  });
  describe('rendering 2 right button', function () {
    it('creates valid snapshot', function () {
      const component = renderer.create(
        <Toolbar
          title={TITLE}
          rightButtons={addRightButtons(MULTI_BUTTON_COUNT)}
          leftButtonType={LeftButtonTypes.menu}
        />,
      );
      expect(component).toMatchSnapshot();
    });
  });

  function addRightButtons(num: number): RightButtonType[] {
    return Array(num).map((_, index) => {
      return {
        title: `${BUTTON_TEMPLATE}${index}`,
        action: () => {},
      };
    });
  }
});
