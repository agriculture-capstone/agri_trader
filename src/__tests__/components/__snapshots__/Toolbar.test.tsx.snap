// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toolbar/> rendering left button renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "flex-start",
        "borderColor": "red",
        "borderWidth": 1,
      },
      Object {
        "backgroundColor": "#4883da",
        "height": 56,
      },
    ]
  }
/>
`;
